packages/core/src/agent/message-list/index.ts

```diff
@@ -2291,9 +2274,11 @@ export class MessageList {
       }
     }
 
-    // 5. Handle text content (fallback if no parts)
-    if (dbMsg.content.content && !hasNonToolReasoningParts) {
-      parts.push({ type: 'text', text: dbMsg.content.content });
+    // 5. Text content is now only stored in parts array
+    // Extract text from parts if needed for fallback
+    const textFromParts = dbMsg.content.parts?.find(p => p.type === 'text')?.text;
+    if (textFromParts && !hasNonToolReasoningParts && !parts.some(p => p.type === 'text')) {
+      parts.push({ type: 'text', text: textFromParts });
     }
```

Comment:
this looks like it's taking all text parts, combining them into one and pushing a duplicate combined text part? definitely remove this